src/main/java/org/opensearch/neuralsearch/processor/util/RestActionUtils.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.util;
+
+import org.opensearch.rest.RestRequest;
+
+import java.util.Optional;
+
+public class RestActionUtils {
+    public static Optional<String[]> splitCommaSeparatedParam(RestRequest request, String paramName) {
```

Comment:
Why we need this class and validations seperately. Similar kind of methods are already being used by opensearch-core. Why not reuse them? Check RestSearchAction in OpenSearch core for the same.